torch/_prims/rng_prims.py

```diff
@@ -0,0 +1,79 @@
+from typing import Tuple
+
+import torch
+from torch import _prims
+from torch._decomp.decompositions_for_rng import RNGStateHelper, throw_on_non_cuda
+
+from torch._prims_common import make_contiguous_strides_for
+from torch._prims_common.wrappers import backwards_not_supported
+from torch.types import _device, _dtype
+
+rngprim_namespace = "rngprims"
+rngprim = torch.library.Library(rngprim_namespace, "DEF")
+rngprim_impl = torch.library.Library(
+    rngprim_namespace, "IMPL", "CompositeExplicitAutograd"
+)
+rngprim_autograd_impl = torch.library.Library(rngprim_namespace, "IMPL", "Autograd")
+rngprim_meta_impl = torch.library.Library(rngprim_namespace, "IMPL", "Meta")
+
+
+def register_philox_rand():
+    name = "philox_rand"
+    schema = "philox_rand(int[] size, Tensor seed, Tensor offset, int[] stride, Device? device=None, ScalarType? dtype=None) -> Tensor"  # noqa: B950
+
+    rngprim.define(schema)
+
+    def _philox_rand_meta(
+        shape: torch.Size,
+        seed: torch.Tensor,
+        offset: torch.Tensor,
+        stride: Tuple[int, ...],
+        device: _device,
+        dtype: _dtype,
+    ):
+        strides = make_contiguous_strides_for(shape)
+        return _prims.TensorMeta(
+            shape=shape, strides=stride, dtype=dtype, device=device
+        )
+
+    def _philox_rand(
+        shape: torch.Size,
+        seed: torch.Tensor,
+        offset: torch.Tensor,
+        stride: Tuple[int, ...],
+        device: _device,
+        dtype: _dtype,
+    ):
+        # stride arg will be useful for distributed usecase. Currently, its unused.
+        if device.type == "cpu":
+            devices = []
+        else:
+            devices = [device]
+
+        if device.type != "cuda":
```

Comment:
IIUC, when `device` is `"cpu"`, this code may throw an exception since `str` object has no `type` attribute.